radicle-node/src/service.rs

```diff
@@ -1018,14 +1046,14 @@ where
                     }
                 }
             }
-            (session::State::Connected { .. }, Message::Fetch { rid }) => {
+            (session::State::Connected { .. }, Message::Fetch { rid, namespaces }) => {
```

Comment:
The namespace isn't needed in the `Fetch` request. There's https://github.com/radicle-dev/heartwood/issues/379 to make that clearer.

Reply:
Seems to me that this is the case of `Receive` in `FetchDirection` , which would have `namespaces` info. No?

Reply:
Yes, but what I'm trying to say is that the uploader doesn't need to know about the namespaces. The `Fetch` message is sent from the fetcher to the uploader, to initiate a fetch; and the uploader doesn't need to know what namespaces are going to be fetched, since that is negotiated by git. Hence, the `Fetch` message doesn't need to contain the namespaces.

Reply:
Maybe it makes more sense looking at the changes :) https://github.com/radicle-dev/heartwood/pull/395